packages/editor/src/components/SvgUploader.tsx

```diff
@@ -0,0 +1,86 @@
+import { FileUploader } from "react-drag-drop-files";
+import toast from "react-hot-toast";
+import { useRecoilState } from "recoil";
+import { fileContentsSelector } from "../state/atoms";
+
+export default function SvgUploader() {
+  const setDomain = useRecoilState(fileContentsSelector("domain"))[1];
+  const setSubstance = useRecoilState(fileContentsSelector("substance"))[1];
+  const setStyle = useRecoilState(fileContentsSelector("style"))[1];
+
+  const handleChange = (svg: File) => {
+    const reader = new FileReader();
+    reader.readAsText(svg);
+    reader.onabort = () => console.log("file reading was aborted");
+    reader.onerror = () => console.log("file reading has failed");
+    reader.onload = () => {
+      const svgText = reader.result?.toString();
+      if (!svgText) {
+        return;
+      }
+
+      const parser = new DOMParser();
+      const xmlDoc = parser.parseFromString(svgText, "text/xml");
+      const styElem = xmlDoc.getElementsByTagName("sty");
```

Comment:
I don't see the `variation` being used here. To faithfully recreate the uploaded SVG, the `variation` needs to be the same as well. You can use `setDiagramMetadata` from `diagramMetadataSelector` to set the metadata.

Reply:
Also bumping this comment